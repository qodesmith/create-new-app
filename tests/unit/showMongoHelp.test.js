const showMongoHelp = require('../../modules/showMongoHelp')


describe('showMongoHelp', () => {
  const originalConsoleLog = console.log
  console.log = jest.fn()

  beforeEach(() => console.log.mockReset())
  afterAll(() => console.log = originalConsoleLog)

  it('should log mongo-specific help to the screen', () => {
    showMongoHelp()
    expect(console.log).toHaveBeenCalledTimes(1)
    expect(console.log.mock.calls).toMatchSnapshot()
  })
})
