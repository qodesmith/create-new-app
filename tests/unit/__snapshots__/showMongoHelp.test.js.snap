// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showMongoHelp should log mongo-specific help to the screen 1`] = `
Array [
  Array [
    "
When deploying apps using MongoDB in production,
you'll want to ensure 2 main things:

  1. You use a different port than the default port (27017).
  2. You use authentication with a username and password to connect.


Wherever you are running MongoDB in production you'll want to
create a new user in the \`admin\` database. SSH into your machine
and you can add a user from the Mongo console like so:

    [1m// Use the \`admin\` database.[22m
    [36muse admin[39m

    [1m// Create a user with the appropriate roles.[22m
    [34mconst[39m [36muser[39m = {
      [36muser:[39m [33m'myUserName'[39m, [90m// Make sure to change this![39m
      [36mpw:[39m [33m'myPassword'[39m, [90m// Make sure to change this![39m
      [36mroles:[39m [
        { [36mrole:[39m [33m'userAdminAnyDatabase'[39m, [36mdb:[39m [33m'admin'[39m },
        [33m'readWriteAnyDatabase'[39m
      ]
    }

    [1m// Save that user to the \`admin\` database.[22m
    [36mdb[39m.[33mcreateUser[39m([36muser[39m)

See [34mhttp://bit.ly/2UTn484[39m for more details.


Mongo-specific variables will be setup in the \`.env\` file.
If you used the guided process, these variables will contain the
[1mdefault values[22m and [1mshould be changed[22m before deploying to production.
You will have to manually set the value to \`MONGO_USER_PASSWORD\`.

If you passed all your options to the cli, these will be filled:

    â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
    â”‚ [36mMONGO_URI[39m                â”‚ [33mmongodb://localhost:<mongoPort>/<appName>[39m     â”‚
    â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
    â”‚ [36mMONGO_URI_PROD[39m           â”‚ [33mmongodb://localhost:<mongoPortProd>/<appName>[39m â”‚
    â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
    â”‚ [36mMONGO_USER[39m               â”‚ [33m<mongoUser>[39m                                   â”‚
    â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
    â”‚ [36mMONGO_USER_PASSWORD[39m      â”‚ [33m<fill this value manually>[39m                    â”‚
    â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
    â”‚ [36mMONGO_AUTH_SOURCE[39m        â”‚ [33m<mongoAuthSource>[39m                             â”‚
    â”œâ”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¼â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”¤
    â”‚ [36mMONGO_SESSION_COLLECTION[39m â”‚ [33m<appName>Sessions[39m                             â”‚
    â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯

When you're ready to deploy your app, make sure the above variables
have the appropriate values. Then copy the \`.env\` file over to your
remote server. [1mIt is git-ignored and will not be in version control[22m.
",
  ],
]
`;
